alg/gdal_interpolateatpoint.cpp

```diff
@@ -124,13 +124,18 @@ bool GDALInterpExtractValuesWindow(GDALRasterBand *pBand,
             // Compose the cached block to the final buffer
             for (int j = 0; j < nLinesToCopy; j++)
             {
-                memcpy(padfAsDouble + ((nFirstLineInOutput + j) * nWidth +
-                                       nFirstColInOutput) *
-                                          nTypeFactor,
-                       poValue->data() +
-                           ((nFirstLineInCachedBlock + j) * nReqXSize +
-                            nFirstColInCachedBlock) *
-                               nTypeFactor,
+                const size_t dstOffset =
+                    ((nFirstLineInOutput + j) * nWidth + nFirstColInOutput) *
```

Comment:
In the unlikely case where someone would try to extract more than 2 billion pixels at at time (will need to run pre-commit to reformat):

```suggestion
                    (static_cast<size_t>(nFirstLineInOutput + j) * nWidth + nFirstColInOutput) *
```

Reply:
done